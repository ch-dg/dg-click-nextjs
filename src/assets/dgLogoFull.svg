<svg width="140" height="48" viewBox="0 0 140 48" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M56.1484 8.17285H60.5435C64.4941 8.17285 67.1114 10.5432 67.1114 14.5926C67.1114 18.642 64.5435 21.0124 60.5929 21.0124H56.1978L56.1484 8.17285ZM60.3954 18.9877C63.0126 18.9877 64.6423 17.4568 64.6423 14.5432C64.6423 11.6296 62.9633 10.1482 60.3954 10.1482H58.4694V18.9877H60.3954Z" fill="#414141"/>
<path d="M74.2714 11.5803C77.0862 11.5803 79.3578 13.605 79.3578 16.4198C79.3578 19.2346 77.0862 21.2593 74.3208 21.2593C71.5554 21.2593 69.2344 19.284 69.2344 16.4198C69.2344 13.5556 71.4566 11.5803 74.2714 11.5803ZM74.2714 19.284C75.8023 19.284 77.0368 18.0988 77.0368 16.568C77.0368 16.5186 77.0368 16.4692 77.0368 16.4198C77.0862 14.889 75.9504 13.605 74.4196 13.5556C72.8887 13.5062 71.6047 14.6421 71.5554 16.1729C71.5554 16.2717 71.5554 16.3211 71.5554 16.4198C71.506 17.9507 72.6912 19.1853 74.1726 19.284C74.222 19.284 74.2714 19.284 74.2714 19.284Z" fill="#414141"/>
<path d="M81.9258 11.7779H84.148V12.8643C84.148 13.0618 84.148 13.3087 84.0986 13.5062H84.148C84.7406 12.3704 85.8764 11.6297 87.1604 11.5803C88.5925 11.5803 89.4813 12.2223 89.827 13.4569H89.8764C90.5184 12.3211 91.7036 11.5803 93.0369 11.5803C94.9628 11.5803 96.0492 12.6667 96.0492 15.0865V21.0124H93.7776V15.5309C93.7776 14.4939 93.5801 13.7038 92.4937 13.7038C91.3085 13.7038 90.5678 14.6421 90.2715 15.7779C90.1727 16.1729 90.1727 16.568 90.1727 16.963V21.0124H87.9011V15.5309C87.9011 14.5433 87.7529 13.7038 86.6665 13.7038C85.4813 13.7038 84.6912 14.642 84.3949 15.8272C84.2962 16.2223 84.2962 16.6174 84.2962 17.0124V21.0618H82.0246V11.7779H81.9258Z" fill="#414141"/>
<path d="M103.16 11.5803C105.827 11.5803 107.259 13.5062 107.259 15.9754C107.259 16.2223 107.21 16.8643 107.21 16.8643H100.839C100.938 18.2964 102.173 19.3828 103.605 19.284C104.593 19.2346 105.531 18.889 106.272 18.2964L107.16 19.926C106.123 20.7655 104.79 21.2593 103.457 21.2593C100.839 21.3581 98.6172 19.3828 98.5185 16.7655C98.5185 16.6667 98.5185 16.568 98.5185 16.4198C98.4691 13.5556 100.444 11.5803 103.16 11.5803ZM104.938 15.284C104.889 14.0495 104.148 13.3581 103.111 13.3581C101.975 13.3087 101.037 14.1482 100.889 15.284H104.938Z" fill="#414141"/>
<path d="M110.123 18.247C110.814 18.889 111.703 19.2347 112.641 19.2347C113.283 19.2347 113.876 18.9384 113.876 18.3952C113.876 17.0125 109.382 17.21 109.382 14.247C109.382 12.5187 110.913 11.531 112.888 11.531C114.024 11.4816 115.111 11.8273 115.95 12.568L115.111 14.2964C114.469 13.7532 113.679 13.5063 112.839 13.5063C112.246 13.5063 111.654 13.7532 111.654 14.3458C111.654 15.6298 116.148 15.4816 116.148 18.4446C116.148 19.9754 114.814 21.21 112.641 21.21C111.308 21.21 110.074 20.7656 109.086 19.8767L110.123 18.247Z" fill="#414141"/>
<path d="M119.161 13.7531H117.976V11.9259H119.21V9.25928H121.482V11.9259H123.556V13.7531H121.482V17.3087C121.482 18.8395 122.667 19.0371 123.309 19.0371C123.457 19.0371 123.605 19.0371 123.704 18.9877V21.0124C123.457 21.0617 123.21 21.0617 123.013 21.0617C121.729 21.0617 119.161 20.6667 119.161 17.605V13.7531Z" fill="#414141"/>
<path d="M126.321 8.17285H128.593V10.1975H126.321V8.17285ZM126.321 11.7778H128.642V21.0124H126.371L126.321 11.7778Z" fill="#414141"/>
<path d="M136.198 11.5803C137.531 11.5309 138.815 12.0247 139.753 12.963L138.766 14.5432C138.124 13.9507 137.235 13.5556 136.346 13.5556C134.815 13.5062 133.58 14.7408 133.531 16.2223C133.531 16.2716 133.531 16.321 133.531 16.4198C133.482 17.9507 134.716 19.284 136.247 19.284C136.296 19.284 136.346 19.284 136.395 19.284C137.432 19.2346 138.37 18.7902 139.111 18.0988L140 19.7284C138.963 20.7161 137.63 21.2593 136.198 21.2099C133.136 21.2099 131.21 19.0371 131.21 16.3704C131.21 13.7037 133.136 11.5803 136.198 11.5803Z" fill="#414141"/>
<path d="M89.7288 28.6174C92.3955 28.6174 93.8276 30.5434 93.8276 33.0125C93.8276 33.3088 93.7782 33.9014 93.7782 33.9014H87.4079C87.5066 35.3335 88.7412 36.4199 90.1733 36.3211C91.1609 36.2718 92.0498 35.9261 92.84 35.3335L93.7288 36.9631C92.6918 37.8026 91.3585 38.2964 90.0251 38.2964C86.9634 38.2964 85.0869 36.0742 85.0869 33.4569C85.0375 30.6421 86.9634 28.6174 89.7288 28.6174ZM91.4572 32.3211C91.4079 31.0866 90.6671 30.3952 89.6301 30.3952C88.4943 30.3458 87.556 31.1853 87.4079 32.3211H91.4572Z" fill="#414141"/>
<path d="M96.2471 28.8642H98.4693V29.9506C98.4693 30.1481 98.4693 30.395 98.4199 30.5926H98.4693C99.1113 29.358 100.395 28.6172 101.778 28.6666C103.803 28.6666 104.988 29.7531 104.988 32.1728V38.0987H102.716V32.6666C102.716 31.5308 102.42 30.7901 101.235 30.7901C100.099 30.7901 99.0619 31.5802 98.7656 32.6666C98.6668 33.0617 98.5681 33.4568 98.6174 33.9012V38.1481H96.3458L96.2471 28.8642Z" fill="#414141"/>
<path d="M111.902 28.6174C114.568 28.6174 116 30.5434 116 33.0125C116 33.3088 115.951 33.9014 115.951 33.9014H109.581C109.679 35.3335 110.914 36.4199 112.346 36.3211C113.334 36.2718 114.223 35.9261 115.013 35.3335L115.902 36.9631C114.865 37.8026 113.531 38.2964 112.198 38.2964C109.136 38.2964 107.26 36.0742 107.26 33.4569C107.21 30.6421 109.136 28.6174 111.902 28.6174ZM113.679 32.3211C113.63 31.0866 112.889 30.3952 111.852 30.3952C110.717 30.3458 109.778 31.1853 109.63 32.3211H113.679Z" fill="#414141"/>
<path d="M118.42 28.8642H120.642V30.4445C120.642 30.7901 120.593 31.0864 120.593 31.0864H120.642C120.988 29.7531 122.173 28.7654 123.556 28.7161C123.704 28.7161 123.803 28.7161 123.951 28.7654V31.0371C123.753 30.9877 123.605 30.9877 123.408 30.9877C122.222 30.9877 121.185 31.7778 120.889 32.9136C120.741 33.4074 120.692 33.9013 120.692 34.3951V38.0494H118.42V28.8642Z" fill="#414141"/>
<path d="M130.766 32.2716H131.062V32.1234C131.062 30.8889 130.321 30.4938 129.284 30.4938C128.395 30.5432 127.506 30.8395 126.766 31.3827L125.877 29.7037C126.914 28.9629 128.198 28.5679 129.482 28.5679C131.901 28.5679 133.334 29.9012 133.334 32.2222V38H131.21V37.2592C131.21 37.0617 131.21 36.8148 131.259 36.6173H131.21C130.667 37.6543 129.58 38.2963 128.395 38.2469C126.766 38.2469 125.284 37.2098 125.284 35.3827C125.284 32.4691 129.235 32.2716 130.766 32.2716ZM128.988 36.5185C130.222 36.5185 131.062 35.2345 131.062 34.0987V33.9012H130.667C129.482 33.9012 127.605 34.0494 127.605 35.3333C127.605 35.9259 128.05 36.5185 128.988 36.5185Z" fill="#414141"/>
<path d="M136.099 25.2593H138.371V34.8889C138.371 35.8766 138.716 36.1729 139.309 36.1729C139.408 36.1729 139.556 36.1729 139.655 36.1729V38.1975C139.408 38.2469 139.161 38.2469 138.914 38.2469C137.63 38.2469 136.05 37.9013 136.05 35.284V25.2593H136.099Z" fill="#414141"/>
<path d="M23.8519 47.8766C10.6667 47.8766 0 37.21 0 23.9754C0 10.7408 10.6667 0.123535 23.8519 0.123535C37.037 0.123535 47.7037 10.7902 47.7037 23.9754C47.7531 37.1606 37.037 47.8766 23.8519 47.8766ZM23.8519 2.44452C12 2.49391 2.37037 12.1235 2.37037 24.0248C2.37037 35.926 12.0494 45.5556 23.9506 45.5063C35.8025 45.5063 45.4321 35.8766 45.4321 23.9754C45.4321 12.1235 35.7531 2.44452 23.8519 2.44452Z" fill="#038AD3"/>
<path d="M40.543 22.8396C40.543 22.642 40.3948 22.4939 40.1479 22.4939H38.0245C37.8269 22.4939 37.6788 22.642 37.6294 22.8396V29.6544H36.6911C34.5183 29.6544 32.7899 28.7655 31.8516 27.284H28.543C29.9257 30.642 33.3331 32.3704 36.6911 32.3704H40.0491C40.1479 32.3704 40.2467 32.3211 40.2961 32.2717C40.3948 32.2223 40.4936 32.0741 40.4442 31.9754V22.8396H40.543Z" fill="#414141"/>
<path d="M7.20996 24.963C7.20996 25.1605 7.35811 25.3087 7.60502 25.358H9.72848C9.92601 25.358 10.0742 25.2099 10.1235 24.963V18.3457H11.0618C13.3334 18.3457 15.0124 19.284 16.0001 20.8642H19.2593C17.8766 17.4074 14.4692 15.6296 11.0618 15.6296H7.65441C7.40749 15.6296 7.25934 15.7778 7.20996 16.0247C7.20996 16.0247 7.20996 16.0247 7.20996 16.0741V24.963Z" fill="#414141"/>
<path d="M15.9507 27.2346C14.8149 28.7655 12.9384 29.7038 11.0124 29.605H7.60502C7.40749 29.605 7.25934 29.7531 7.20996 29.9507V30.9383V31.926C7.20996 32.1235 7.35811 32.2717 7.60502 32.321H11.0618C15.0124 32.321 18.1729 30.2963 19.3581 27.2346H15.9507Z" fill="#414141"/>
<path d="M31.7037 20.8642C32.8395 19.2346 34.7161 18.2963 36.6914 18.3457H40.1482C40.3457 18.3457 40.4938 18.1975 40.5432 18V16.0247C40.5432 15.8272 40.3951 15.679 40.1482 15.6296H36.6914C32.642 15.6296 29.4815 17.7037 28.3457 20.8642H31.7037Z" fill="#414141"/>
<path d="M80.6416 32.2222V36.0741H78.9626C76.3947 36.0741 74.7157 34.5432 74.7157 31.679C74.7157 28.8148 76.3947 27.2346 78.9626 27.2346H82.37V25.2593H78.8144C74.8638 25.2593 72.2959 27.6296 72.2959 31.679C72.2959 35.7284 74.8638 38.0988 78.8144 38.0988H82.9132V32.2222H80.6416Z" fill="#414141"/>
<path d="M30.5184 22.6914H25.1851V17.3086C25.1851 17.1111 25.037 16.963 24.79 16.9136H22.9135C22.716 16.9136 22.5678 17.0617 22.5184 17.3086V22.642H17.1851C16.9876 22.642 16.8394 22.7901 16.79 23.037V24.9136C16.79 25.1111 16.9382 25.2593 17.1851 25.3086H22.5184V30.642C22.5184 30.8395 22.6666 30.9876 22.9135 31.037H24.79C24.9876 31.037 25.1357 30.8889 25.1851 30.642V25.3086H30.5184C30.716 25.3086 30.8641 25.1605 30.9135 24.9136V23.037C30.9135 22.8395 30.7653 22.6914 30.5184 22.6914Z" fill="#038AD3"/>
<path d="M58.0249 31.1852C56.9879 30.642 56.3459 29.5556 56.4447 28.3704C56.4447 26.1482 58.2225 25.0618 60.4447 25.0618C60.9879 25.0618 61.5805 25.1605 62.0743 25.2593L61.5311 27.0865C61.2348 27.0371 60.9879 26.9877 60.6916 26.9877C59.6052 26.9877 58.7657 27.5803 58.7657 28.6667C58.7657 29.4074 59.1607 30.3457 60.6916 30.3457H62.7163V28.7161H64.9879V30.3457H66.6175V32.321H64.9879V33.8025C64.9879 36.3704 63.3089 38.0988 60.4447 38.0988C57.7286 38.0988 55.9015 36.4692 55.9015 34.2963C55.8521 32.9136 56.6916 31.6791 58.0249 31.1852ZM60.3459 36.0247C61.778 36.0247 62.6175 35.2346 62.6175 33.7037V32.321H60.4941C59.1114 32.321 58.1731 32.963 58.1731 34.1976C58.2225 35.1852 59.062 36.0247 60.3459 36.0247Z" fill="#038AD3"/>
</svg>
