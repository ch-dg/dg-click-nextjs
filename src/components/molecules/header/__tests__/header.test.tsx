import { render, screen } from "@testing-library/react";
import dgLogoFull from "@/assets/dgLogoFull.svg";
import Header from "@/components/molecules/header/header";

describe("Unit:Molecule header", () => {
  const mockPrimaryLogoBar = {
    image: {
      src: dgLogoFull,
      alt: "domestic and general logo",
      className: "w-auto h-auto",
      priority: true,
    },
    bgClassName: "px-6 md:px-12 bg-white",
    className: "flex items-center justify-start py-6",
  };

  const mockComponent = (
    <Header
      className="mb-24 flex w-screen flex-col justify-center"
      primaryLogoBar={mockPrimaryLogoBar}
    />
  );

  it("renders LogoBar", () => {
    render(mockComponent);
    const primaryImage = screen.getByRole("img", {
      name: /domestic and general logo/i,
    });

    expect(primaryImage).toBeInTheDocument();
    expect(primaryImage).toHaveAttribute("alt", "domestic and general logo");
    expect(primaryImage).toHaveClass("w-auto h-auto");
  });

  it("checks against snapshot", () => {
    const view = render(mockComponent);
    expect(view).toMatchSnapshot();
  });
});
