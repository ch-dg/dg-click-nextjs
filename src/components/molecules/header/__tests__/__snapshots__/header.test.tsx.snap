// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit:Molecule header checks against snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mb-24 flex w-screen flex-col justify-center"
      >
        <div
          class=""
        >
          <div
            class="container mx-auto"
          >
            <div
              class="ml-[12%] w-auto h-auto"
            >
              <img
                alt="primary"
                class="ml-[12%] w-auto h-auto"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
            </div>
          </div>
        </div>
        <div
          class=""
        >
          <div
            class="container mx-auto"
          >
            <div
              class="ml-[12%] w-auto h-auto"
            >
              <img
                alt="secondary"
                class="ml-[12%] w-auto h-auto"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mb-24 flex w-screen flex-col justify-center"
    >
      <div
        class=""
      >
        <div
          class="container mx-auto"
        >
          <div
            class="ml-[12%] w-auto h-auto"
          >
            <img
              alt="primary"
              class="ml-[12%] w-auto h-auto"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
          </div>
        </div>
      </div>
      <div
        class=""
      >
        <div
          class="container mx-auto"
        >
          <div
            class="ml-[12%] w-auto h-auto"
          >
            <img
              alt="secondary"
              class="ml-[12%] w-auto h-auto"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
